linux/userdata.sh

```diff
@@ -1,174 +1,70 @@
-    done
-
-    if [[ "$${ERREXIT}" == "1" ]]
-    then
-        set -e
-    fi
-
-    return $result
-}  # ----------  end of function retry  ----------
-
-finally() {
-  local exit_code="$${1:-0}"
-
-  echo "Finally: "
-
-  # everything to happen whether install succeeds or fails
-
-  # write the status to a file for reading by test script
-  printf "%s\n" "$${userdata_status[@]}" > /tmp/userdata_status
-
-  # allow ssh to be on non-standard port (SEL-enforced rule)
-  setenforce 0
-
-  # open firewall (iptables for rhel/centos 6, firewalld for 7
-  systemctl status firewalld &> /dev/null
-  if [ $? -eq 0 ] ; then
-    echo "Configuring firewalld..."
-    firewall-cmd --zone=public --permanent --add-port=122/tcp
-    firewall-cmd --reload
-  else
-    echo "Configuring iptables..."
-    iptables -A INPUT -p tcp --dport 122 -j ACCEPT #open port 122
-    service iptables save
-    service iptables restart
-  fi
-
-  sed -i -e '5iPort 122' /etc/ssh/sshd_config
-  sed -i -e 's/Port 22/#Port 22/g' /etc/ssh/sshd_config
-  cat /etc/ssh/sshd_config
-  service sshd restart
-
-  # create a directory with all the build artifacts
-  artifact_base="/tmp/terrafirm"
-  artifact_dir="$${artifact_base}/build-artifacts"
-  mkdir -p "$${artifact_dir}/scap_output"
-  mkdir -p "$${artifact_dir}/cloud"
-  cp -R /var/log/watchmaker/ "$${artifact_dir}"
-  cp -R /root/scap/output/* "$${artifact_dir}/scap_output/"
-  cp -R /var/log/cloud*log "$${artifact_dir}/cloud/"
-
-  # move logs to s3
-  s3_keyfix=$(cat /etc/redhat-release | cut -c1-3)$(cat /etc/redhat-release | sed 's/[^0-9.]*\([0-9]\.[0-9]\).*/\1/') # get OS version as key prefix
-  artifact_location="s3://${tfi_s3_bucket}/${tfi_build_date}/${tfi_build_hour}_${tfi_build_id}/$${s3_keyfix}"
-  echo "Writing logs to $${artifact_location}"
-  cp "${tfi_lx_userdata_log}" "$${artifact_dir}"
-  aws s3 cp "$${artifact_dir}" "$${artifact_location}" --recursive || true
-
-  # creates compressed archive to upload to s3
-  zip_file="$${artifact_base}/${tfi_build_date}-${tfi_build_id}-$${s3_keyfix}.tgz"
-  cd "$${artifact_dir}"
-  tar -cvzf "$${zip_file}" .
-  aws s3 cp "$${zip_file}" "s3://${tfi_s3_bucket}/${tfi_build_date}/${tfi_build_hour}_${tfi_build_id}/" || true
-
-  exit "$${exit_code}"
-}
-
-catch() {
-  local this_script="$0"
-  local exit_code="$${1:-1}"
-  local err_lineno="$2"
-  echo "$0: line $2: exiting with status $${exit_code}"
-
-  userdata_status=($exit_code "Userdata install error at stage $stage")
-
-  finally $@
-}
-
 # setup error trap to go to catch function
 trap 'catch $? $${LINENO}' ERR
 
-# everything below this is the TRY
-
-# start time of install
-start=`date +%s`
-
-# declare an array to hold the status (number and message)
-userdata_status=(0 "Success")
+if [[ "$ami_key" == *pkg ]]; then
+  # if it ends with 'pkg', test standalone
+
+  standalone_location="s3://${tfi_s3_bucket}/${tfi_build_date}/${tfi_build_hour}_${tfi_build_id}/release/latest/watchmaker-latest-standalone-linux-x86_64"
```

Comment:
I'm not seeing where we use these vars individually. Would it be simpler to define a single var here and combine them on the terraform side? Maybe `${build_slug}` or `${s3_bucket_base}` or something?